http-core/src/main/scala/org/apache/pekko/http/impl/engine/http2/hpack/ByteStringInputStream.scala

```diff
@@ -24,13 +27,29 @@ import pekko.util.ByteString.ByteString1C
 @InternalApi
 private[http2] object ByteStringInputStream {
 
-  def apply(bs: ByteString): InputStream =
-    bs match {
-      case cs: ByteString1C =>
-        // TODO optimise, ByteString needs to expose InputStream (esp if array backed, nice!)
-        new ByteArrayInputStream(cs.toArrayUnsafe())
-      case _ =>
-        // NOTE: We actually measured recently, and compact + use array was pretty good usually
-        apply(bs.compact)
+  private lazy val byteStringInputStreamMethodTypeOpt = Try {
+    val lookup = MethodHandles.publicLookup()
+    val inputStreamMethodType = MethodType.methodType(classOf[InputStream])
+    lookup.findVirtual(classOf[ByteString], "asInputStream", inputStreamMethodType)
+  }.toOption
```

Comment:
1. I think we should drop the `lazy` and just make it a `val`.
2. the `toOption` will introduce an allocation, use `OptionVal`

Reply:
this code is only called once in the whole lifetime of the JVM

Reply:
I think it should be lazy because this code may need to run unless it is needed

Reply:
But the `lazy` will introduce a `volatile`  iirc, I think this is a trade-off, just not sure if that will cause the problem in the native image.

Reply:
The getorelse will introduce another closure creation, think we can just fallback to imperative style for performance reason

Reply:
I found that the getOrElse has a small cost so I made a change to remove it